pkg/controller/tsp/metric.go

```diff
@@ -0,0 +1,132 @@
+package tsp
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/gocrane/api/prediction/v1alpha1"
+)
+
+func (c *Controller) RegisterMetric() {
```

Comment:
should we have tsp controller implement the metric collector interface, I think we should have a metrics struct instead.

Reply:
ok, i will refactor it to a metric collector